app/Http/Controllers/Admin/CategoryController.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Category;
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use App\Http\Requests\CategoryRequest;
+
+class CategoryController extends Controller
+{
+    public function add()
+    {
+        return view('admin.category.add');
+    }
+
+    public function store(CategoryRequest $request)
+    {
+        $validated = $request->validated();
+        $created_category = Category::create([
+            'title'        => $validated['title'],
+            'description'  => $validated['description']
+        ]);
+
+        return redirect('/admin/home')->with('my_status', __('Create category is succeeded!'));
+    }
+
+    public function edit($id)
+    {
+        $category = Category::where('id', $id)->first();
+        return view('admin.category.edit', compact('category'));
+    }
+
+    public function update(CategoryRequest $request, $id)
+    {
+        $validated = $request->validated();
+        Category::where('id', $id)
+            ->update([
+                'title' => $validated['title'],
+                'description'  => $validated['description']
+            ]);
+
+        return redirect('/admin/home')->with('my_status', __('Edit category is succeeded!'));
+    }
+
+    public function delete($id)
```

Comment:
We have a technique called type-hinting and also Model binding. What it does in your parameters you can declare the type as Model like this:

```
public function edit (User $id) {
..
}
```

In this code it will automatically inject `User` model in method's parameter. So we can perform model functions inside the block no need to do `Model::find()` function. Also advantage is the `$id` will only strictly accept Model object, so if the parameter is not a model then it will send error automatically.

```
public function update(Category $category)
{
    $category->update();
}
```

Reply:
I changed function to use type-hinting.